consensus/helpers.go

```diff
@@ -317,3 +314,18 @@ func (m *consensusModule) getValidatorsAtHeight(height uint64) ([]*coreTypes.Act
 
 	return persistenceReadContext.GetAllValidators(int64(height))
 }
+
+func (m *consensusModule) IsValidator() (bool, error) {
```

Comment:
This seems very inefficient, but also not the primary goal of the PR.

Can you add a TODO that would do one of two things:
1. Single query to the DB in the persistence module
2. Cache it in the `consensus` module